pkg/rtc/room.go

```diff
@@ -526,6 +526,38 @@ func (r *Room) UpdateSubscriptions(
 }
 
 func (r *Room) SyncState(participant types.LocalParticipant, state *livekit.SyncState) error {
+	pLogger := participant.GetLogger()
+	pLogger.Infow("setting sync state", "state", state)
+
+	shouldReconnect := false
+	pubTracks := state.GetPublishTracks()
+	existingPubTracks := participant.GetPublishedTracks()
+	for _, pubTrack := range pubTracks {
+		// client may not have sent TrackInfo for each published track
+		ti := pubTrack.Track
+		if ti == nil {
+			pLogger.Warnw("TrackInfo not sent during resume", nil)
+			shouldReconnect = true
+			break
+		}
+
+		found := false
+		for _, existingPubTrack := range existingPubTracks {
+			if existingPubTrack.ID() == livekit.TrackID(ti.Sid) {
```

Comment:
Checking only for SID match. Should this check other stuff?

Reply:
It would be ok that the Sid must return from sfu side